etcdserver/etcdhttp/client_security.go

```diff
@@ -0,0 +1,267 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdhttp
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+
+	"github.com/coreos/etcd/etcdserver"
+	"github.com/coreos/etcd/etcdserver/etcdhttp/httptypes"
+	"github.com/coreos/etcd/etcdserver/security"
+)
+
+type securityHandler struct {
+	store       *security.Store
+	clusterInfo etcdserver.ClusterInfo
+}
+
+func handleSecurity(mux *http.ServeMux, server *etcdserver.EtcdServer) {
+	sec := securityHandler{
+		store:       security.NewStore(server, defaultServerTimeout),
+		clusterInfo: server.Cluster,
+	}
+	mux.HandleFunc(securityPrefix+"/roles", sec.baseRoles)
+	mux.HandleFunc(securityPrefix+"/roles/", sec.handleRoles)
+	mux.HandleFunc(securityPrefix+"/users", sec.baseUsers)
+	mux.HandleFunc(securityPrefix+"/users/", sec.handleUsers)
+}
+
+func (sec *securityHandler) handleRoles(w http.ResponseWriter, r *http.Request) {
+	subpath := path.Clean(r.URL.Path[len(securityPrefix):])
+	// Split "/roles/rolename/command".
+	// First item is an empty string, second is "roles"
+	pieces := strings.Split(subpath, "/")[2:]
+	switch len(pieces) {
+	case 1:
+		sec.forRole(w, r, pieces[0])
+	case 2:
+		sec.actionRole(w, r, pieces[0], pieces[1])
+	}
+}
+
+func (sec *securityHandler) baseRoles(w http.ResponseWriter, r *http.Request) {
+	if !allowMethod(w, r.Method, "GET") {
+		return
+	}
+	w.Header().Set("X-Etcd-Cluster-ID", sec.clusterInfo.ID().String())
+	w.Header().Set("Content-Type", "application/json")
+	var rolestruct struct {
```

Comment:
rolesCollections? i want to make sure we use the consistent naming across the codebase. @bcwaldon has got the membersCollections before you.